### PR TITLE
feat: [CDS-77113]: do not color logs coming from stderr

### PR DESCRIPTION
**Pull request by tmacari-harness**
## Describe your changes
[CDS-77113]: do not color logs coming from stderr
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

[CDS-77113]: https://harness.atlassian.net/browse/CDS-77113?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/53148)

### CODE REVIEW

**Comment by tmacari-harness** [comment]
trigger uts

**Review by abosii-harness — COMMENTED** [review]
Based on customer complain it's not only color but the tag as well, are we handling it somewhere? And what about the actual errors, how removing coloring from stderr is going to help customer improve readability of the logs? It's just changing it from red to white I guess

**Comment by tmacari-harness** [comment]
We always add tag to the logs,, and those tags can be hidden from UI is uses do not want to see them.
Would be incorrect to add INFO tag on logs coming from stdout and nothing on logs coming form stderr (or add INFO tag to them).
That's what customer was complaining about that logs that are not error logs are printed with red color.
Terraform and terragrunt are adding some formatting on the logs on their own (colouring something, printing in specific formats...). We will display the same formatting in UI execution logs.

**Comment by tmacari-harness** [comment]
trigger uts

**Comment by tmacari-harness** [comment]
trigger uts

**Comment by vlica-harness** [comment]
I think the rollback steps is missing for terraform/terragrunt @tmacari-harness

File: 125-cd-nextgen/src/main/java/io/harness/cdng/provision/terraform/TerraformPlanStepV2.java [hunk 1]
```diff
@@ -353,6 +355,8 @@ private TerraformTaskNGParametersBuilder getTerraformTaskNGParametersBuilder(
         .timeoutInMillis(
             StepUtils.getTimeoutMillis(stepElementParameters.getTimeout(), TerraformConstants.DEFAULT_TIMEOUT))
         .useOptimizedTfPlan(true)
+        .skipColorLogs(featureFlagHelper.isEnabled(accountId, CDS_TF_TG_SKIP_ERROR_LOGS_COLORING))
+
```
Comment (vlica-harness) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/provision/terraform/TerraformPlanStepV2.java:359]: minor: can we remove the empty line

File: 930-delegate-tasks/src/test/java/io/harness/delegate/task/terragrunt/TerragruntApplyTaskNGTest.java [hunk 2]
```diff
@@ -91,30 +94,13 @@ public void testApplyRunModule() throws JoseException, IOException, InterruptedE
...
-        cliHelper.executeCliCommand(
-            eq("terragrunt plan -out=tfplan -input=false -target=\"test-target\" -var-file=\"test-terragrunt-12345.tfvars\" "),
-            anyLong(), eq(applyParameters.getEnvVars()), any(), any(), any(), any(), any(), anyLong()))
-        .thenReturn(CliResponse.builder().exitCode(0).build());
-    when(cliHelper.executeCliCommand(eq("terragrunt apply -input=false -lock-timeout=10s tfplan"), anyLong(),
-             eq(applyParameters.getEnvVars()), any(), any(), any(), any(), any(), anyLong()))
-        .thenReturn(CliResponse.builder().exitCode(0).build());
-    when(cliHelper.executeCliCommand(contains("terragrunt output -json >"), anyLong(), eq(applyParameters.getEnvVars()),
-             any(), any(), any(), any(), any(), anyLong()))
+    when(cliHelper.executeCliCommand(any()))
```
Comment (vlica-harness) [inline 930-delegate-tasks/src/test/java/io/harness/delegate/task/terragrunt/TerragruntApplyTaskNGTest.java:97]: @tmacari-harness by above code, we made sure we mock exactly and we know exactly what commands are called.
Now with the refactoring, it's not quite clear what we mock. The test gets not very useful.

**Comment by tmacari-harness** [comment]
trigger featurenamecheck

**Comment by tmacari-harness** [comment]
trigger uts

**Review by tmacari-harness — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/53148)* status: 49 file(s) to review. 1. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`930-delegate-tasks/src/test/java/io/harness/delegate/task/terragrunt/TerragruntApplyTaskNGTest.java` line 97 at r1](https://reviewable.io/reviews/harness/harness-core/53148#gh-1333935541:-Nevo2zCD8eP9cVLWrAr:b-ijyivf) ([raw file](https://github.com/harness/harness-core/blob/81463f87ae4b46929802390c090ffa20d071e893/930-delegate-tasks/src/test/java/io/harness/delegate/task/terragrunt/TerragruntApplyTaskNGTest.java#L97)):*
<details><summary><i>Previously, vlica-harness (Vasile Lica) wrote…</i></summary><blockquote>

@tmacari-harness by above code, we made sure we mock exactly and we know exactly what commands are called.
Now with the refactoring, it's not quite clear what we mock. The test gets not very useful.
</blockquote></details>

Will check with verify that executeCliCommand was called with expected params

**Comment by tmacari-harness** [comment]
trigger uts

File: 930-delegate-tasks/src/test/java/io/harness/delegate/task/terragrunt/TerragruntApplyTaskNGTest.java (hunk 2 above)
Comment (tmacari-harness) [reply 930-delegate-tasks/src/test/java/io/harness/delegate/task/terragrunt/TerragruntApplyTaskNGTest.java:96]: fixed

File: 125-cd-nextgen/src/main/java/io/harness/cdng/provision/terraform/TerraformPlanStepV2.java (hunk 1 above)
Comment (tmacari-harness) [reply 125-cd-nextgen/src/main/java/io/harness/cdng/provision/terraform/TerraformPlanStepV2.java]: Done

**Comment by tmacari-harness** [comment]
trigger uts

**Comment by tmacari-harness** [comment]
trigger uts

**Review by vlica-harness — DISMISSED** [review]

**Comment by tmacari-harness** [comment]
trigger uts

**Review by vlica-harness — APPROVED** [review]

**Review by abosii-harness — APPROVED** [review]

**Comment by tmacari-harness** [comment]
trigger review

**Pull request merged**